lib/mongo/operation/shared/polymorphic_result.rb

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2018 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  module Operation
+
+    # Shared behavior of instantiating a result class matching the
+    # operation class.
+    #
+    # @api private
+    module PolymorphicResult
+      private
+
+      def result_class
+        bits = self.class.name.sub(/::[^:]*$/, "::Result").split('::')
+        cls = Object
```

Comment:
Not necessary to change if you prefer it this way, but this looks like it's equivalent to `bits.reduce(Object) { |cls, name| cls.const_get(name) }`

Reply:
Thanks, I've been looking for a shorter way to express that.